src/compiler/checker.ts

```diff
@@ -11485,13 +11484,12 @@ namespace ts {
                         // such that direct inferences to T get priority over inferences to Partial<T>, for example.
                         const inference = getInferenceInfoForType((<IndexType>constraintType).type);
                         if (inference && !inference.isFixed) {
-                            const key = (source.symbol ? getSymbolId(source.symbol) + "," : "") + getSymbolId(target.symbol);
-                            if (contains(mappedTypeStack, key)) {
+                            const key = (source.symbol ? getSymbolId(source.symbol) + "s" : "") + getSymbolId(target.symbol);

```

Comment:
Normally we distinguish different _types_ of cache keys with a leading character, rather than a different separator, see `getLiteralType` for prior art. Plus, I think something like `T123,12` vs `S11,14` is going to make more sense in the debugger than `123,12` vs `11s14`.

Reply:
Sure. Done.